### PR TITLE
feat: add Alert component

### PR DESCRIPTION
**Pull request by adamperrry**
This PR adds an Alert component per Tyson's design (to be initially used in the Connect Modal redesign). The alert is optionally dismissible via props, and the optional `show` and `onDismiss` props allow for even greater control over dismissing/showing the alert.

image

### CODE REVIEW

File: src/components/alerts/Alert/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		container,
+		children,
+		...otherProps
+	} = props;
+
+	const [showAlert, setShowAlert] = useState(show);
+
+	const handleDismiss = () => {
+		setShowAlert(false);
+		if (onDismiss) onDismiss();
```
Comment (tihal) [inline src/components/alerts/Alert/Alert.tsx:38]: I'm rusty on my component lifecycle, but will this cause console errors to the effect of "attempting to call function within an unmounted component"?

**Review by tihal — COMMENTED** [review]
Just one Q before approval!

File: src/components/alerts/Alert/Alert.tsx (hunk 1 above)
Comment (adamperrry) [reply src/components/alerts/Alert/Alert.tsx:38]: I don't think so - tested it in story book with a console.log, and it registers

Comment (tihal) [reply src/components/alerts/Alert/Alert.tsx]: Sweet!

**Review by tihal — APPROVED** [review]
Good to go!

**Review by tihal — APPROVED** [review]
Approved!

**Pull request merged**